editor/app/src/main/AndroidManifest.xml

```diff
@@ -9,13 +9,27 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme" >
+
         <activity android:name=".MainActivity" >
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <provider
+            android:authorities="ca.ubc.eml.soiltopargraphy.editor.fileProvider"
+            android:name="android.support.v4.content.FileProvider"
+            android:exported="false"
+            android:grantUriPermissions="true">
+            <meta-data
+                android:name="android.support.FILE_PROVIDER_PATHS"
+                android:resource="@xml/file_paths"/>
+        </provider>
     </application>
-
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
```

Comment:
Rather than do all the image loading in the app, I believe you can you Intents so that we don't need all these permissions for selecting an image from the phone camera or gallery. I found some [boilerplate code in this StackOverflow answer](https://stackoverflow.com/questions/10165302/dialog-to-pick-image-from-gallery-or-from-camera).